app/controllers/bookmarks_controller.rb

```diff
@@ -43,7 +43,9 @@ def update
   end
   
   def destroy
-    @bookmark = Bookmark.find(params[:id])
+    @user = current_user
+    @topic = @user.topics.find(params[:id])
+    @bookmark = @topic.bookmarks.find(params[:id])
     
```

Comment:
Good protection - `edit`, `update` and `destroy` now protect bookmarks from being modified by non-owners 👍